h5py/tests/test_sliced_arrays.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+import h5py
+from h5py import File
+from .common import TestCase
+
+
+class TestSlicedArrays(TestCase):
+    """Sliced Arrays are stored correctly"""
```

Comment:
Minor naming nitpick: we specifically want to test a non-contiguous array. Slicing with a step > 1 is the easiest way to get that, but a slice like `arr[5:10]` wouldn't show it.

Also, I think this can be added in an existing file - maybe in the class `h5py.tests.test_dataset.TestVlen`.